checkov/openapi/runner.py

```diff
@@ -13,11 +18,22 @@ def import_registry(self) -> BaseCheckRegistry:
         from checkov.openapi.checks.registry import openapi_registry
         return openapi_registry
 
-    def _parse_file(self, f: str) -> None:
-        raise Exception("parser should be implemented")
+    def _parse_file(self, f: str) -> tuple[dict[str, Any] | list[dict[str, Any]], list[tuple[int, str]]] | tuple[None, None]:
+        if f.endswith(".json"):
+            return JsonRunner._parse_file(self, f)
```

Comment:
before returning the json or yaml file results we need to check it's indeed an open API file and not any yaml (for example not a cfn yaml). for that you'll need to either validate the file schema or look for an element that is always there. if it's not an open API spec you should return None,None.

in github actions it is solved by inspecting path: https://github.com/bridgecrewio/checkov/blob/master/checkov/github_actions/runner.py#L24

in cloudformation by inspecting an element 
https://github.com/bridgecrewio/checkov/blob/master/checkov/cloudformation/parser/cfn_yaml.py#L218

in github vcs by inspecting a schema:
https://github.com/bridgecrewio/checkov/blob/0c1e05c6ac94d4fd8957b1b5a1cb4334b6e1fa0f/checkov/github/base_github_org_security.py#L23

schema is the most accurate. Does openapi have a json schema you can validate or an element you check for existence?

Reply:
maybe this can be used to validate a file is indeed a json schema: https://github.com/OAI/OpenAPI-Specification/blob/main/schemas/v3.0/schema.json

Reply:
sounds good.